uaclient/api/u/pro/services/list/v1.py

```diff
@@ -0,0 +1,206 @@
+# Services List
+import logging
+from typing import Dict, List, Optional
+
+from uaclient import exceptions, util
+from uaclient.api.api import APIEndpoint
+from uaclient.api.data_types import AdditionalInfo
+from uaclient.api.u.pro.status.is_attached.v1 import _is_attached
+from uaclient.config import UAConfig
+from uaclient.contract import get_available_resources
+from uaclient.data_types import (
+    BoolDataValue,
+    DataObject,
+    Field,
+    StringDataValue,
+    data_list,
+)
+from uaclient.entitlements import entitlement_factory
+from uaclient.entitlements.base import UAEntitlement
+from uaclient.entitlements.entitlement_status import ContractStatus
+from uaclient.files import machine_token
+
+LOG = logging.getLogger(util.replace_top_level_logger_name(__name__))
+
+
+class ServiceInfo(DataObject, AdditionalInfo):
+    fields = [
+        Field("name", StringDataValue, doc="Name of the service"),
+        Field(
+            "description", StringDataValue, doc="Description of the service"
+        ),
+        Field(
+            "available",
+            BoolDataValue,
+            doc="Service availability for a subscription",
+        ),
+        Field(
+            "entitled",
+            BoolDataValue,
+            doc="Entitlement of a service for a subscription",
+        ),
+    ]
+
+    def __init__(
+        self,
+        *,
+        name: str,
+        desc: str,
+        available: bool,
+        entitled: Optional[bool] = None
+    ):
+        self.name = name
+        self.description = desc
+        self.available = available
+        self.entitled = entitled
+
+
+class ServiceListResult(DataObject, AdditionalInfo):
+    fields = [
+        Field("services", data_list(ServiceInfo), doc="List of services"),
+    ]
+
+    def __init__(self, *, services: List[ServiceInfo]):
+        self.services = services
+
+
+def get_service_list() -> ServiceListResult:
```

Comment:
generally the public function name has matched the final word in the endpoint string. So to follow that pattern, this would be `list()`